src/Microsoft.AspNet.Mvc.ViewFeatures/ViewExecutor.cs

```diff
@@ -36,7 +36,20 @@ public static class ViewExecutor
         {
             var response = actionContext.HttpContext.Response;
 
-            contentType = contentType ?? DefaultContentType;
+            if (contentType == null)
+            {
+                contentType = DefaultContentType;
+
+                if (!string.IsNullOrEmpty(response.ContentType))
+                {
+                    MediaTypeHeaderValue parsedContentType;
+                    if (MediaTypeHeaderValue.TryParse(response.ContentType, out parsedContentType))
+                    {
+                        contentType = parsedContentType;
+                    }
```

Comment:
Why do we need to do this? Why does MVC care if you set an invalid content type?